pyfolio/plotting.py

```diff
@@ -1344,3 +1344,63 @@ def show_worst_drawdown_periods(returns, top=5):
     drawdown_df['net drawdown in %'] = list(
         map(utils.round_two_dec_places, drawdown_df['net drawdown in %']))
     print(drawdown_df.sort('net drawdown in %', ascending=False))
+
+
+def plot_monthly_returns_timeseries(returns, ax=None, **kwargs):
+    """
+    Plots monthly returns as a timeseries.
+
+    Parameters
+    ----------
+    returns : pd.Series
+        Daily returns of the strategy, noncumulative.
+         - See full explanation in tears.create_full_tear_sheet.
+    ax : matplotlib.Axes, optional
+        Axes upon which to plot.
+    **kwargs, optional
+        Passed to seaborn plotting function.
+
+    Returns
+    -------
+    ax : matplotlib.Axes
+        The axes that were plotted on.
+    """
+
+    if ax is None:
+        ax = plt.gca()
+
+    monthly_ret_table = timeseries.aggregate_returns(returns, 'monthly')
+    monthly_ret_table = monthly_ret_table.reset_index()
+    monthly_ret_table.columns = ['year', 'month', 'returns']
+
+    # Generate month-year labels for the x-axis corresponding to
+    # the returns plotted
+    date_labels = [str(i[1]) + ' - ' + str(i[0]) for i in
```

Comment:
yep that's right, the defaults are super ugly because it prints the month/day/year of each rendered monthly bar and the plot becomes illegible if more than ~30 bars. 
As well, the defaults include the last business _day_ of the month in each label (which just adds to the cluttered visual because the last business day of each month jumps around, 29, 30, 31, etc). My opinion is that including the _day_ in the label is not really proper when labeling a monthly value, as well.
Pandas does not have a monthly frequency unit so I could not up-sample the DateTimeIndex to make it more elegant (which would have been best).
Given the above, and my desire to have an easily interpreted plot, how would you suggest improved pruning of the labels? I actually adapted my implementation from an accepted StackOverflow response
**Sent from iPhone**

    _____________________________
From: Thomas Wiecki <notifications@github.com>
Sent: Wednesday, November 4, 2015 4:10 AM
Subject: Re: [pyfolio] ENH implement plot_monthly_returns_timeseries in plotting.py to rende… (#195)
To: quantopian/pyfolio <pyfolio@noreply.github.com>
Cc: Justin Lent <justinlent@gmail.com>


    

In pyfolio/plotting.py:   > +    Returns> +    -------> +    ax : matplotlib.Axes> +        The axes that were plotted on.> +    """> +> +    if ax is None:> +        ax = plt.gca()> +> +    monthly_ret_table = timeseries.aggregate_returns(returns, 'monthly')> +    monthly_ret_table = monthly_ret_table.reset_index()> +    monthly_ret_table.columns = ['year', 'month', 'returns']> +> +    # Generate month-year labels for the x-axis corresponding to> +    # the returns plotted> +    date_labels = [str(i[1]) + ' - ' + str(i[0]) for i in   

What's wrong with the default labels of this plot? I suppose there's too many? How about just thinning those. The current approach feels a bit brittle.   

—
Reply to this email directly or view it on GitHub.